### PR TITLE
Cadmium + LazyDFU port

### PR DESCRIPTION
**Pull request by Titaniumtown**
I still may need to add code comments and such. But this is a port of the 1.16 port of cadmium.

I also ported LazyDFU.

### CODE REVIEW

File: patches/server/0066-Cadmium.patch [hunk 1]
```diff
@@ -0,0 +1,1177 @@
...
+-import com.mojang.datafixers.DataFixUtils;
+-import com.mojang.datafixers.FieldFinder;
+-import com.mojang.datafixers.FunctionType;
+-import com.mojang.datafixers.OpticFinder;
+-import com.mojang.datafixers.RewriteResult;
+-import com.mojang.datafixers.TypeRewriteRule;
+-import com.mojang.datafixers.Typed;
+-import com.mojang.datafixers.TypedOptic;
+-import com.mojang.datafixers.View;
++import com.mojang.datafixers.*;
```
Comment (toinouH) [inline patches/server/0066-Cadmium.patch]: `import com.mojang.datafixers.*;` Are you sure about the * import ?

File: patches/server/0066-Cadmium.patch [hunk 2]
```diff
@@ -0,0 +1,1177 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Simon Gardling <titaniumtown@gmail.com>
```
Comment (toinouH) [inline patches/server/0066-Cadmium.patch]: Shouldn't it be the email of LucilleTea (and you as co-author ?)

**Review by toinouH — COMMENTED** [review]
You forgot all "Yatopia - start" and "Yatopia- end" stuff.
Isn't there any OBFHELPER ?

**Comment by Titaniumtown** [comment]
@toinouH I am aware that I have to do those things, they're on my plate to do.

File: patches/server/0066-Cadmium.patch (hunk 2 above)
Comment (Titaniumtown) [reply patches/server/0066-Cadmium.patch]: Fixed

File: patches/server/0067-Port-LazyDFU.patch [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+--- a/src/main/java/net/minecraft/server/DataConverterRegistry.java
++++ b/src/main/java/net/minecraft/server/DataConverterRegistry.java
+@@ -18,7 +18,8 @@ public class DataConverterRegistry {
+     // Yatopia start
+     private static final DataFixer c;
+     static {
+-        DataFixerBuilder datafixerbuilder = new DataFixerBuilder(SharedConstants.getGameVersion().getWorldVersion());
++        // DataFixerBuilder datafixerbuilder = new DataFixerBuilder(SharedConstants.getGameVersion().getWorldVersion());
++        DataFixerBuilder datafixerbuilder = new me.steinborn.lazydfu.mod.LazyDataFixerBuilder(SharedConstants.getGameVersion().getWorldVersion());
+
```
Comment (duplexsystem) [inline patches/server/0067-Port-LazyDFU.patch]: Remove commented out line and add //Yatopia comment

**Review by duplexsystem — CHANGES_REQUESTED** [review]

File: patches/server/0067-Port-LazyDFU.patch (hunk 3 above)
Comment (Titaniumtown) [reply patches/server/0067-Port-LazyDFU.patch]: Fixed.

**Review by duplexsystem — APPROVED** [review]

**Comment by duplexsystem** [comment]
Add LazyDFU and cadmium to readme

**Comment by Titaniumtown** [comment]
> Add LazyDFU and cadmium to readme

Done.

**Comment by Titaniumtown** [comment]
I think there is an issue with LazyDFU + ishland's patches, further testing needed.

**Comment by Titaniumtown** [comment]
Ok, it seems cadmium actually causes issues with chunk gen with threaded world gen. I'll be closing this PR until that issue can be resolved.

**Comment by Titaniumtown** [comment]
Re-opening this because it turns out the issue persists with ishland's world gen patches even without these patches.

**Review by duplexsystem — APPROVED** [review]

**Comment by Titaniumtown** [comment]
Merging as these patches have been being tested for a long while now without any issues, more testing is needed though, hence why it's going in the staging branch.

**Pull request merged**